executor/slow_query.go

```diff
@@ -118,16 +116,48 @@ func (e *slowQueryRetriever) close() error {
 	return nil
 }
 
-func (e *slowQueryRetriever) dataForSlowLog(ctx sessionctx.Context) ([][]types.Datum, error) {
-	reader := bufio.NewReader(e.files[e.fileIdx].file)
-	rows, err := e.parseSlowLog(ctx, reader, 1024)
+type parsedSlowLog struct {
+	rows [][]types.Datum
+	err  error
+}
+
+func (e *slowQueryRetriever) parseDataForSlowLog(ctx context.Context, sctx sessionctx.Context) {
+	for e.fileIdx < len(e.files) {
+		reader := bufio.NewReader(e.files[e.fileIdx].file)
```

Comment:
The `bufio.Reader` can be reused during the processing, and switch underlying files via [bufio.Reader.Reset](https://golang.org/pkg/bufio/#Reader.Reset) method.

Reply:
Thanks for your suggestion! I have changed it.